include/deal.II/multigrid/multigrid.h

```diff
@@ -74,6 +74,60 @@ class Multigrid : public Subscriptor
     f_cycle
   };
 
+  /**
+   * A structure that has boost::signal objects for a number of actions that
+   * happen for the typical usage of this class.
+   *
+   * For documentation on signals, see
+   * http://www.boost.org/doc/libs/release/libs/signals2 .
+   */
+  struct Signals
+  {
+    /**
+     * This signal is triggered before and after the call to
+     * MGTransfer::copy_to_mg which transfers given vector to all levels.
+     */
+    boost::signals2::signal<void (int level, bool start)> transfer_to_mg;
+
+    /**
+     * This signal is triggered before and after the call to
+     * MGTransfer::copy_from_mg which transfers a multi-level vector
+     * to a normal vector.
+     */
+    boost::signals2::signal<void (int level, bool start)> transfer_to_global;
+
+    /**
+     * This signal is triggered before and after the call to the coarse solver.
+     */
+    boost::signals2::signal<void (int level, bool start)> coarse_solve;
+
+    /**
+     * This signal is triggered before and after the call to
+     * MGTransfer::restrict_and_add() which restricts a vector from @p level
+     * to the next coarser one.
```

Comment:
maybe explicitly say `@p level - 1`?

Reply:
This should be correct.

Reply:
it is, sure, i just though we can remind readers what "coarser" means in terms of level number